WordPress/Classes/ViewRelated/Blog/My Site/NoSitesView.swift

```diff
@@ -0,0 +1,188 @@
+import SwiftUI
+
+protocol NoSitesViewDelegate: AnyObject {
+    func didTapAccountAndSettingsButton()
+}
+
+struct AddNewSiteConfiguration {
+    let canCreateWPComSite: Bool
+    let canAddSelfHostedSite: Bool
+    let launchSiteCreation: () -> Void
+    let launchLoginForSelfHostedSite: () -> Void
+}
+
+struct NoSitesView: View {
+
+    @SwiftUI.Environment(\.colorScheme) var colorScheme
+
+    @State private var isShowingDialog = false
+
+    let viewModel: NoSitesViewModel
+    let addNewSiteConfiguration: AddNewSiteConfiguration
+
+    weak var delegate: NoSitesViewDelegate?
+
+    var body: some View {
+        ZStack(alignment: .bottom) {
+            Color(uiColor: .listBackground)
+                .edgesIgnoringSafeArea(.all)
+
+            makeMainView()
+                .padding(.horizontal, 32)
+
+            if viewModel.isShowingAccountAndSettings {
+                makeAccountAndSettingsButton()
+                    .padding(.horizontal, 32)
+                    .padding(.bottom, 20)
+            }
+        }
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+    }
+
+    private func makeMainView() -> some View {
+        VStack(alignment: .leading, spacing: 16) {
```

Comment:
Even though the progress is on hold on Design System, I've added a file named `Length` where you can find virtually all CGFloat options we ought to use for padding, margin, spacing etc. Certainly not required my any means but I wanted to mention its existance:

`Length.Padding.double` would return a `CGFloat` of value 16.0.

Reply:
For this PR I decided not to use `Length` because there's a mixture of 8x and 10x metric values. Going forward I'll keep `Length` in mind - thank you Alp!

Reply:
Just want to note that it could make sense to check with designers if their 10x usage were intentional. Because we had talked about this before with them and afaik they both use the 8x values. I wonder if there's a reason for mixing them together.

Reply:
Updated! https://github.com/wordpress-mobile/WordPress-iOS/pull/21456/commits/768e1889d18274e3c5ff38a9090e4b89db7e83e4

Slack ref: p1693497168975339/1693307309.040859-slack-C04SFL0RP51